sdjwt/src/test/kotlin/web5/security/SdJwtTest.kt

```diff
@@ -0,0 +1,196 @@
+package web5.security
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.nimbusds.jose.JOSEObjectType
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.JWSHeader
+import com.nimbusds.jose.JWSSigner
+import com.nimbusds.jose.crypto.impl.ECDSAProvider
+import com.nimbusds.jose.jwk.JWK
+import com.nimbusds.jose.util.Base64URL
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import web5.sdk.crypto.InMemoryKeyManager
+import web5.sdk.crypto.KeyManager
+import web5.sdk.dids.DidKey
+import java.io.File
+import java.io.IOException
+
+val example1 = File("src/test/resources/example1.sdjwt").readText()
+
+class SdJwtTest {
+  private val mapper = jacksonObjectMapper().apply {
+    enable(SerializationFeature.INDENT_OUTPUT)
+    setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    setDefaultPrettyPrinter(CustomPrettyPrinter())
+  }
+
+  @Test
+  fun `parse and serialize are inverse functions`() {
+    val sdJwt = SdJwt.parse(example1)
+    val serialized = sdJwt.serialize(mapper)
+    assertEquals(example1, serialized)
+  }
+
+  @Test
+  fun `test the mapper serializes objects as expected`() {
+    val result = mapper.writeValueAsString(
+      mapOf(
+        "street_address" to "123 Main St",
+        "locality" to "Anytown",
+      )
+    )
+    assertEquals("""{"street_address": "123 Main St", "locality": "Anytown"}""", result)
+  }
+
+  @Test
+  fun `whole flow from issuer to holder to verifier`() {
+    // First issue an SD-JWT
+    val blinder = SdJwtBlinder()
+
+    val claimsData = """{      
+      "given_name": "Mister",
+      "family_name": "Tee",
+      "birthdate": "1940-10-31",
+      "nationalities": ["US", "DE", "PO"],
+      "address": {
+        "street": "happy street 123",
+        "zip_code": "12345"
+      }
+    }""".trimIndent()
+    val claimsToBlind = mapOf(
```

Comment:
Why can't this be dynamic? Is it to give the option to the user? 
If they provide null as claimsToBlind can it dynamically do default blinding to all?

Reply:
The issuer of an SdJwt must, at some point, select which claims they'll allow the holder to selectively disclose. For those, they must also say how they'll be disclosable, since there are various mechanisms.

What do you think should be the way in which they are all blinded by default?